Formula/pyenv.rb

```diff
@@ -47,6 +49,20 @@ def install
   end
 
   test do
-    shell_output("eval \"$(#{bin}/pyenv init -)\" && pyenv versions")
+    # Create a fake python version and executable.
+    shell_output("mkdir -p \"$(pyenv root)/versions/1.2.3/bin\"")
+    shell_output("echo 'echo hello' > \"$(pyenv root)/versions/1.2.3/bin/foo\"")
+    shell_output("chmod +x \"$(pyenv root)/versions/1.2.3/bin/foo\"")
```

Comment:
I'm not quite sure what you're doing here, but, as the name suggests, `shell_output` isn't really meant to be used for running arbitrary shell commands. Try using Ruby/`brew` methods here. For example:
```ruby
mkdir_p "path/to/create"
```
Note also that the test runs in a sandbox, so I don't think you'll be able to write where you intend to in this test.